include/seastar/core/metrics_types.hh

```diff
@@ -82,6 +83,17 @@ struct histogram {
      */
     histogram operator+(histogram&& h) const;
 
+    // Everything below here is for native histograms (also known as sparse histograms).
+    // Native histograms are an experimental Prometheus feature.
+
+    // schema defines the bucket schema. Currently, valid numbers are -4 <= n <= 8.
+    // They are all for base-2 bucket schemas, where 1 is a bucket boundary in each case, and
+    // then each power of two is divided into 2^n logarithmic buckets.
+    // Or in other words, each bucket boundary is the previous boundary times 2^(2^-n).
+    // In the future, more bucket schemas may be added using numbers < -4 or > 8.
+    std::optional<int32_t> schema;
+    // min_id is the first bucket id of a given schema.
+    std::optional<int32_t> min_id;
```

Comment:
suggest: struct histogram_schema, and make the whole struct member optional.  This avoids ambiguity if one is set and the other is missing.

Reply:
I like the idea, will change